internal/film/filmservice_mongo.go

```diff
@@ -25,5 +30,50 @@ func (s FilmMongoService) GetAllFilm() ([]types.Film, error) {
 	return s.repo.GetAllFilm()
 }
 func (s FilmMongoService) AddFilm(f types.Film) (types.Film, error) {
+	if f.ID != "" {
+		f.ID = crawler.NewUUID()
+	}
 	return s.repo.InsertFilm(f)
 }
+func (s FilmMongoService) GenerateFilm() error {
+	l := flog.New()
+	l.SetLocal("film")
+	filmIn := make(chan types.Film, 20)
+	filmOut := make(chan types.Film, 20)
+	go func() {
+		var wg sync.WaitGroup
+		for i := 1; i <= 20; i++ {
+			wg.Add(1)
+			go crawler.ExtractDetail(i, &wg, filmIn, filmOut, l)
+		}
+		wg.Wait()
+	}()
+	go func() {
+		for {
+			select {
+			case film, ok := <-filmOut:
+				if !ok {
+					l.Infof("all data is saved")
+					return
+				}
+				if film.ID != "" {
+					film.ID = crawler.NewUUID()
+				}
+				_, err := s.repo.InsertFilm(film)
+				if err != nil {
+					l.Errorf("can't save data to database %v", err)
+				}
+				l.Infof("%s is saved", film.Title)
+			}
+		}
+	}()
+	l.Infof("extraction is stared")
+	err := crawler.MakeURLTopRate(filmIn, l)
+	if err != nil {
+		return err
+	}
+	l.Infof("extraction is done close channel")
+	defer close(filmIn)
+	defer close(filmOut)
```

Comment:
This is bad.
You still sending to `filmOut` inside `ExtractDetail`.
So what if this function already finished (`filmOut` closed) and the goroutine `ExtractDetail` still running and sending data to `filmOut`?
Panic, right?